software/py/blood_graph.py

```diff
@@ -37,282 +39,322 @@ class BloodGraph:
     # for generating the key
     _KEY_WIDTH  = 512
     _KEY_HEIGHT = 512
-    _DEFAULT_START_CYCLE = 50000 
-    _DEFAULT_END_CYCLE   = 250000
 
-    # default constructor
-    def __init__(self, timing_stats_file, abstract):
-
-        self.timing_stats_file = timing_stats_file
-        self.abstract = abstract
 
-        # List of types of stalls incurred by the core 
-        self.stalls_list   = ["stall_depend_remote_load_dram",
-                              "stall_depend_local_remote_load_dram",
-                              "stall_depend_remote_load_global",
-                              "stall_depend_remote_load_group",
-                              "stall_depend_local_remote_load_global",
-                              "stall_depend_local_remote_load_group",
-                              "stall_lr_aq",
-                              "stall_depend",
-                              "stall_depend_local_load",
-                              "stall_fp_local_load",
-                              "stall_fp_remote_load",
-                              "stall_force_wb",
-                              "stall_icache_store",
-                              "stall_remote_req",
-                              "stall_local_flw",
-                              "stall_amo_aq",
-                              "stall_amo_rl",
-                              "icache_miss",
-                              "stall_ifetch_wait",
-                              "bubble_icache",
-                              "bubble_branch_mispredict",
-                              "bubble_jalr_mispredict",
-                              "bubble_fp_op",
-                              "bubble",
-                              "stall_md" ]
+    # List of types of stalls incurred by the core 
+    _STALLS_LIST   = ["stall_depend_remote_load_dram",
```

Comment:
Good. This is a better design pattern. What motivated you to do it?

Reply:
I suppose my coding style improved a bit over time thanks to your comments on my PR's over time :D when I came back to iterate over this code, there seemed to be many things wrong with it that I didn't see before.